### DISCUSSION TITLE
How to handle error when sending an event

### DISCUSSION THREAD

**Discussion by andywpt** [statelyai/xstate#4549] [Q&A]
When sending an event, there are two types of error to consider:

**Case 1. The event is allowed in the current state but does not pass some guard statement. 
Case 2. The event is not allowed in the current state.**

In both cases, is there a way to pass out the error so I can do some error handling ?  Something like the following:
```
try {
    send "ACCEPT_ORDER" event
} catch {
    // order is in pending state, but courier can only accept one order at a time (Case 1)
    error "already accepted another order"  

   // order is not in pending state (Case 2)
    error "the order has been accepted by others"
}
  ```

**Comment by Andarist** [comment]
What would you like to happen in both of those scenarios? Could you illustrate this with some pseudocode?

**Reply by andywpt** [reply]
@Andarist Sure, I've updated my question.
